src/main/kotlin/de/smartsquare/starter/mqtt/MqttPublisher.kt

```diff
@@ -37,17 +43,96 @@ class Mqtt5Publisher(private val adapter: MqttMessageAdapter, client: Mqtt5Clien
 
     private val asyncClient = client.toAsync()
 
+    /**
+     * Options for publishing a message.
+     */
+    data class PublishingOptions(
+        val retain: Boolean = false,
+        val messageExpiryInterval: Long = MqttPublish.NO_MESSAGE_EXPIRY,
+        val payloadFormatIndicator: Mqtt5PayloadFormatIndicator? = null,
+        val contentType: String? = null,
+        val responseTopic: MqttTopic? = null,
+        val correlationData: ByteArray? = null,
+        val userProperties: Mqtt5UserProperties = Mqtt5UserProperties.of(),
+    ) {
+        companion object {
+            /**
+             * Creates a new [Builder] for [PublishingOptions]. With the given [original] as a base.
+             */
+            @JvmStatic
+            @JvmOverloads
+            fun builder(original: PublishingOptions = PublishingOptions()) = object : Builder {
```

Comment:
Could you please re-check here? I also added a fluent builder for Java clients.
I think this makes sense, as Java clients can't rely on the syntactic sugar Kotlin can offer with it's `copy` method and default args. As a Java client you can now build options in the following way:

```kotlin
val baseOptions = PublishingOptions.builder()
    .contentType("text/plain")
    .messageExpiryInterval(10)
    .build()
val options = PublishingOptions.builder(baseOptions)
    .messageExpiryInterval(20)
    .build()
```

Reply:
Very nice idea, I like it :+1: